### DISCUSSION TITLE
Individual variable value loading from a file

### DISCUSSION THREAD

**Discussion by David-Lor** [pydantic/pydantic#4794] [Feature Request]
I've got a feature request that may be useful in certain use-cases, mainly regarding settings management using Pydantic classes.

When deploying a service using Docker Swarm, one can use Docker Secrets to pass sensible settings such as passwords or tokens to the deployed application's settings. Each secret can be read from within the container on a certain path, as a simple text file.

The idea of this feature is to allow defining a special value on individual variables, which will make their values being read from a file specified on their original value.

Example code:

```python
import pydantic

class MySettings(pydantic.BaseModel):
    host: str
    port: int
    password: pydantic.SecretStr

    class Config:
        file_support = True

settings = MySettings(host="localhost", port=8080, password="file:/run/secrets/my_secret")
```

In this example, the value from `password` is read from a file located in `/run/secrets/my_secret`.

If this looks a good/interesting implementation for Pydantic, I could try submitting a PR. Otherwise, I'd create a library that can be used for making pydantic classes compatible with this behaviour.

Regards.

**Comment by nymous** [comment]
Hello! The `BaseSettings` class already supports reading from a file (e.g. Docker secrets), see this docs https://pydantic-docs.helpmanual.io/usage/settings/#secret-support
Is this what you have in mind?

If you need something else more custom, you can always [implement your own "settings source"](https://pydantic-docs.helpmanual.io/usage/settings/#customise-settings-sources), like I did in [Pydantic-Vault](https://github.com/nymous/pydantic-vault).

**Reply by David-Lor** [reply]
Thanks for the hint! I was unaware of this feature, I actually searched for "setting values from files" but didn't imagine Pydantic would already have support for Docker Secrets :smile_cat: 

Anyway I think it may be more useful being able to customize the secrets' location, so I may go ahead with my own implementation.
